hazelcast/src/main/java/com/hazelcast/sql/impl/schema/TableResolver.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2008-2020, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.schema;
+
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Generic interface that resolves tables belonging to a particular backend.
+ * <p>
+ * At the moment the interface does exactly what we need - provides tables and registers default search paths.
+ * In future, if we have more objects to expose, it might be expanded or reworked completely.
+ */
+public interface TableResolver {
+    /**
+     * @return Search paths to be added for object resolution.
+     */
+    List<List<String>> getDefaultSearchPaths();
```

Comment:
Why it's returning list of lists?

Reply:
A search path is a list of components. E.g. search patch 'hazelcast.partitioned' is represented as List("hazelcast", "partitioned"). Therefore, if there are multiple search paths, then this is a list of lists.

Reply:
I see, probably it makes sense to explain that in javadoc.

Reply:
BTW, does order of elements in collections returned from `getDefaultSearchPaths` and `getTables` make some difference in resolution?

Reply:
Yes, the order is important - if two objects with the same name exist in two default search paths, we will pick the object from the path that goes earlier. I'll add this to JavaDocs.

Reply:
Added JavaDocs.